src/CalcManager/Ratpack/exp.cpp

```diff
@@ -392,6 +392,31 @@ void powratNumeratorDenominator(PRAT *px, PRAT y, uint32_t radix, int32_t precis
     destroyrat(pxPow);
 }
 
+void powratNumeratorDenominator(PRAT *px, PRAT y, uint32_t radix, int32_t precision)
+{
+    // We need px as simple as possible
+    gcdrat(px, precision);
```

Comment:
Looking at the code for the underlying gcd() function, there are two assumptions made about the inputs:
1. Inputs are in BASEX
2. Inputs are integers

Here, the powrat* functions can be called with a different radix than BASEX, however, which would cause gcdrat to fail. Debugging this, however, shows that CalcInput::ToRational always generates a BASEX Rational, so this ends up working out.

I think there are more issues to dig into around how radices are used and assumptions made around them, but for the purposes of this change, could you see if the fix still works if you:
1. Pass the radix to gcdrat as a parameter
2. Inside gcdrat, if radix != BASEX, convert `a` to BASEX by calling numtonRadixx() on a->pp and a->pq (this is what I haven't checked: does numtonRadixx work if you SAY radix is 10, when in fact the PNUMBER passed is in BASEX?)
3. Perform the gcd calculation
4. Convert output back into provided radix (if not BASEX) using nRadixxtonum and return (this will likely not actually work right now, since the passed in values will actually be BASEX due to CalcInput::ToRational)